### ISSUE TITLE
Backend

### ISSUE THREAD

**Issue by pepellou** [refu-gal/meanit#11]
Reemplazar el prototipo en firebase por un backend básico que persista necesidades y mensajes de contacto. Habría que desplegarlo en algún sitio también (¿heroku?).

**Comment by adoankim** [comment]
Me parece correcto, así evitamos que los datos de configuración sensibles estén en el repo y simplificamos la capa de vista.

¿Se mantendría Firebase para la persistencia?

Y sobre el sitio de despliegue. En Heroku, si empezamos con la versión gratuita, tendremos el inconveniente de que la instancia se pone a dormir tras un periodo de inactividad. Se podría ver algún servicio que no tenga esta restricción.

**Comment by pepellou** [comment]
Me suena que en firebase se pueden poner reglas por dominio, lo que permitiría que nadie use la BD aun con la apiKey. Podría ser una solución a corto plazo para el tema de la seguridad.

De hacer un backend, creo que la tecnología es indiferente (las necesidades iniciales son muy típicas). @adoankim ¿alguna predilección? Igual la tarea no es muy prioritaria, pero estaría bien antes de hacer nueva funcionalidad.

**Comment by adoankim** [comment]
Mmm tiene mucho sentido, lo de Firebase se puede controlar por ahí para evitarnos apuros. 

Sobre la tecnología, pienso igual, cualquiera nos puede valer con que no requiera mucha complejidad para gestionar lo que tenemos.

**Comment by pepellou** [comment]
En el telegram de DevGalicia @carlisgg sugirió hacer una quedada en Santiago para programar un miniproyecto desde cero usando "react-redux-router", podría ser una opción para programar, aprender y quizás captar algún programador más. Es una idea.

Hoy le echaré un ojo a lo de los permisos por dominio en firebase. Si lo dejamos así, esta issue se puede hacer con calma, por ejemplo esperando al evento que propone @carlisgg 😬

**Comment by adoankim** [comment]
ok, suena genial!. Pero eso sería más para la parte del cliente web, ¿no?.

**Comment by pepellou** [comment]
Cierto. Yo es que de React no tengo ni idea, asumo que haremos algún tipo de backend en node o así. Era una idea por abrir el proyecto a la comunidad, también podemos pasar de @carlisgg sencillamente 🤣

**Comment by adoankim** [comment]
Por lo contrario, me parece genial incluir algo más mantenible en el lado cliente 😄. Sólo quería comentar lo referente al backend. Y si, cuanta más gente participe mejor!, de eso se trata, no?

**Comment by pepellou** [comment]
:100: Totalmente de acuerdo.
Cuando lo tengamos más claro habrá que redefinir esta issue seguramente.
A ver si @carlisgg recibe las notificaciones y organizamos :slightly_smiling_face:

**Comment by pepellou** [comment]
En firebase sólo se puede filtrar por dominio con OAuth activado, es decir, habría que tener un login. Probé la opción de "anonymous provider" que en teoría hace OAuth anónimo transparente al usuario, pero no me da opciones.

No sé, podríamos poner un login (con firebase podemos poner fácilmente login con FB, twitter, google, github e email+pass), pero habría que ponerlo tanto para refugiados como para voluntarios (esto último no es lo que hablamos, aunque por otro lado tampoco parece tan grave).

**Comment by adoankim** [comment]
mmm creo que seguiríamos teniendo el problema de permisos. Creo que lo de Firebase deberíamos tenerlo de lado backend, puesto que mísmamente con OAuth, alguien con el token del proyecto seguiría teniendo permiso a la bd de Firebase.

Lo deal es que Firebase permitiese CORS 😅

**Comment by pepellou** [comment]
Sí, sí, lo ideal es un backend. El tema es que en teoría firebase te permite, cuando tienes un "sign in" method, hacer una lista blanca de request providers para filtrar de qué dominios se aceptan peticiones (es decir, desde otro dominio, aun con el token, te fallaría). Eso es lo que leí en la documentación al menos, no lo probé.

**Comment by adoankim** [comment]
@pepellou ok, eso nos valdría perfectamente. Igual podríamos comentar un poco que cosas hacen falta para poder ir trabajando en ellas :).

A nivel de backend:
 - Proyecto Node.JS base para poder desplegarlo en algún PaaS. Algo básico con Express nos llegaría, ¿no?
 - Integrar "sign in" con Firebase para filtrar el acceso. ¿Estaría el backend de por medio o se gestionaría directamente desde cliente?
 - Definir los endpoints para la petición de ayudas y listado de ayudas

¿Se te ocurre alguna cosa más?.
